### PR TITLE
[IMPROVEMENT] Introduce the enumType in ConfigOptions

### PR DESCRIPTION
**Pull request by zuston**
### What changes were proposed in this pull request?
Introduce the enumType in ConfigOptions

### Why are the changes needed?
When I want to extend the zstd compression type in uniffle, I found we lack the enum class for configOptions. 
And I think it will be used for users to specify the concrete implementation in some pluggable class.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
UTs.

### CODE REVIEW

File: common/src/test/java/org/apache/uniffle/common/config/ConfigOptionTest.java [hunk 1]
```diff
@@ -56,6 +56,34 @@ public void testSetKVWithStringTypeDirectly() {
...
+    assertEquals(TestType.TYPE_1, conf.get(enumConfigOption));
+
+    // case2: return the user specified value
+    conf.set(enumConfigOption, TestType.TYPE_2);
+    assertEquals(TestType.TYPE_2, conf.get(enumConfigOption));
+
+    // case3: set enum val with string
+    conf = new RssBaseConf();
+    conf.setString("rss.enum", "TYPE_2");
+    assertEquals(TestType.TYPE_2, conf.get(enumConfigOption));
```
Comment (jerqi) [inline common/src/test/java/org/apache/uniffle/common/config/ConfigOptionTest.java:84]: Should we add a case about wrong type?

**Comment by jerqi** [comment]
cc @smallzhongfeng , the pr is useful for your pr https://github.com/apache/incubator-uniffle/pull/192.

**Comment by smallzhongfeng** [comment]
> cc @smallzhongfeng , the pr is useful for your pr #192.

Good job, this is helpful for my pr. After you review this PR, I will repair my current pr.

**Review by jerqi — APPROVED** [review]
LGTM, thanks @zuston

**Pull request merged**